backend-rust/src/indexer.rs

```diff
@@ -466,22 +486,20 @@ impl ProcessEvent for BlockProcessor {
         let start_time = Instant::now();
         let mut out = format!("Processed {} blocks:", batch.len());
         let mut tx = self.pool.begin().await.context("Failed to create SQL transaction")?;
-        let mut override_context = None;
-        for data in batch {
-            let context = override_context.as_ref().unwrap_or(self.current_context.borrow());
-            let new_context = data.save(context, &mut tx).await.context("Failed saving block")?;
-            if new_context.is_some() {
-                override_context = new_context;
+
+        let new_context =
+            PreparedBlock::batch_save(batch, self.current_context.borrow(), &mut tx).await?;
```

Comment:
It is still done atomically since the entire batch and the block items are part of the same database transaction, so if something fails, we roll back the entire batch.
Note that the batch size will be 1 when the indexer is caught up to the chain.